### PR TITLE
Wait more for newlog and vlans

### PR DESCRIPTION
**Pull request by giggsoff**
Tests for RPi (ROL) indicate that our timeouts for newlog and vlans do not cover real workloads (for slowest case with RPi running xen and zfs).

### CODE REVIEW

File: pkg/newlog/newlogd-init.sh [hunk 1]
```diff
@@ -1,29 +1,43 @@
...
-    LOOP_COUNT=$((LOOP_COUNT + 1))
-    NEWLOGD_PID=$(cat /run/newlogd.pid)
+    sleep 3
+    LOOP_COUNT=$((LOOP_COUNT + 3))
 done
 
+if [ ! -f "$NEWLOGD_TOUCH_FILE" ]; then
+    echo "$(date -Ins -u) gave up waiting for $NEWLOGD_TOUCH_FILE"
+else
+    echo "$(date -Ins -u) waited $LOOP_COUNT for $NEWLOGD_TOUCH_FILE"
```
Comment (zededa-yuri) [inline pkg/newlog/newlogd-init.sh:19]: NIT: `waited $LOOP_COUNT seconds`

File: pkg/newlog/newlogd-init.sh [hunk 2]
```diff
@@ -1,29 +1,43 @@
 #!/bin/sh
 
+echo "$(date -Ins -u) starting newlogd"
 /usr/bin/newlogd &
-mkdir -p /run/watchdog/pid
-touch /run/watchdog/pid/newlogd.pid
+NEWLOGD_PID=$!
+echo "$(date -Ins -u) newlogd is starting..., pid $NEWLOGD_PID"
```
Comment (zededa-yuri) [inline pkg/newlog/newlogd-init.sh:6]: NIT: here and other invocations of `data` - for readability it's always better to use full form of arguments in the scripts
```
--iso-8601=ns --utc
```

File: pkg/newlog/newlogd-init.sh [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
+NEWLOGD_TOUCH_FILE="/run/newlogd.touch"
 LOOP_COUNT=0
-while [ -z "$NEWLOGD_PID" ] && [ "$LOOP_COUNT" -le 30 ];
+while [ ! -f "$NEWLOGD_TOUCH_FILE" ] && [ "$LOOP_COUNT" -le 60 ];
 do
-    sleep 1
-    LOOP_COUNT=$((LOOP_COUNT + 1))
-    NEWLOGD_PID=$(cat /run/newlogd.pid)
+    sleep 3
+    LOOP_COUNT=$((LOOP_COUNT + 3))
```
[the remark this answers is not in the document]
Comment (eriknordmark) [reply pkg/newlog/newlogd-init.sh]: +1

[the remark this answers is not in the document]
Comment (giggsoff) [reply pkg/newlog/newlogd-init.sh]: Actually it comes from [device-steps](https://github.com/lf-edge/eve/blob/f86565b8e77cb832e73a27f7beaec5ae326064a2/pkg/pillar/scripts/device-steps.sh#L47-L61). I added another message about in progress of waiting and will show it every 3 seconds as well.

File: pkg/newlog/newlogd-init.sh (hunk 2 above)
Comment (giggsoff) [reply pkg/newlog/newlogd-init.sh]: Good point, modified. Another interesting thing here is that busybox's date from alpine 3.13 do not support `-Ins`, only `-Is` so I was forced to add `coreutils` package.
```
BusyBox v1.32.1 () multi-call binary.

Usage: date [OPTIONS] [+FMT] [TIME]

Display time (using +FMT), or set time

	[-s,--set] TIME	Set time to TIME
	-u,--utc	Work in UTC (don't convert to local time)
	-R,--rfc-2822	Output RFC-2822 compliant date string
	-I[SPEC]	Output ISO-8601 compliant date string
			SPEC='date' (default) for date only,
			'hours', 'minutes', or 'seconds' for date and
			time to the indicated precision
	-r,--reference FILE	Display last modification time of FILE
	-d,--date TIME	Display TIME, not 'now'
	-D FMT		Use FMT (strptime format) for -d TIME conversion

Recognized TIME formats:
	hh:mm[:ss]
	[YYYY.]MM.DD-hh:mm[:ss]
	YYYY-MM-DD hh:mm[:ss]
	[[[[[YY]YY]MM]DD]hh]mm[.ss]
	'date TIME' form accepts MMDDhhmm[[YY]YY][.ss] instead
```

Comment (zededa-yuri) [reply pkg/newlog/newlogd-init.sh]: >  forced to add coreutils package

Yes, I ran into this earlier too. Adding coreutils is the right way to solve that, I think

**Comment by Insei** [comment]
Previously, in the switch_net_vlans test on rpi4, an error was very likely to occur, now 10 out of 10 switch_net_vlans tests pass without problems.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Pull request merged**